src/Engine/Combat/CombatEngine.java

```diff
@@ -109,11 +111,12 @@ public static BattleSummary calculateBattleResults(UnitContext attacker, UnitCon
 
   public static double calculateOneStrikeDamage( Unit attacker, int battleRange, Unit defender, GameMap map, int terrainStars, boolean attackerMoved )
   {
-    UnitContext attackerContext = new UnitContext(map, attacker, null, attacker.x, attacker.y);
+    XYCoord dest = new XYCoord(attacker);
+    UnitContext attackerContext = new UnitContext(map, attacker, null, Utils.findShortestPath(attacker, dest, map), dest);
```

Comment:
This didn't need a path before... seems strange to require one now.

Reply:
I think I want to start putting paths into combat stuff in general, since path-dependence is a pretty common thing among custom CO concepts.

Reply:
Like Wargroove's knights with their "Do more damage if you move x spaces" mechanic? That's... uh. I mean. I guess.

I generally prefer to add new things at time of need though, otherwise YAGNI.

Reply:
YAGNI doesn't apply here IMO, because this is a data coherency issue and also the sort of thing that could introduce subtle inconsistencies.

Let's say, for example, that I add a CO that gains +50 base damage based on pathing.
The AI is contemplating an infantry that's in range of an enemy tank.
Muriel uses `calculateOneStrikeDamage()`, which leaves out pathing info, and determines that attacking the tank is a bad idea.
Wally uses the full combat logic, including the path, and discovers a very cost-effective trade.
Where's the bug? (assuming, for the sake of argument, that +50 base damage isn't a terrible idea)

Reply:
> the sort of thing that could introduce subtle inconsistencies

Yes. This is. Because we didn't pass a path into `calculateOneStrikeDamage`, we invented one, which may not be the same as what the user entered. Want subtle bugs? This is how you get them.

> Where's the bug?

No bug in the scenario you gave. Muriel just isn't making full use of available tactical information.

Actually, not true; there is a bug, since Wally could find an interesting path that grants extra damage, and then we `findShortestPath` instead of using the carefully crafted non-shortest path, and Wally does an attack expecting a buff and then doesn't get it.

Reply:
> Yes. This is. Because we didn't pass a path into `calculateOneStrikeDamage`, we invented one, which may not be the same as what the user entered. Want subtle bugs? This is how you get them.

That's a very good point, and why I was vaguely intending to have callers pass in `UnitContext`s to `CombatEngine`, for the most part. I'll go fix that.